internal/pkg/source/source.go

```diff
@@ -4,47 +4,143 @@ import (
 	"bufio"
 	"bytes"
 	"errors"
-	"fmt"
 	"io"
+	"os"
 
 	"github.com/hedhyw/json-log-viewer/internal/pkg/config"
 )
 
 const (
 	maxLineSize = 8 * 1024 * 1024
-
-	logEntriesEstimateNumber = 256
 )
 
-// ParseLogEntriesFromReader reads the input and parses all logs.
-func ParseLogEntriesFromReader(
-	reader io.Reader,
-	cfg *config.Config,
-) (LazyLogEntries, error) {
-	reader = io.LimitReader(reader, cfg.MaxFileSizeBytes)
+type Source struct {
```

Comment:
suggestion: let's describe in comments how it works, the implementation is complex.